### PR TITLE
[ENG-4814] creator search

### PR DESCRIPTION
**Pull request by aaxelb**
-   Ticket: [ENG-4814]
-   Feature flag: n/a

## Purpose
allow searching by creator

## Summary of Changes
- add `creator.name` to the properties used for free-text search (depends on https://github.com/CenterForOpenScience/SHARE/pull/812 )
- skip showing creators without a text query
- when searching within a filter facet, display affiliations
- fix a z-index bug so dropdown content is visible atop the dialog box

## Screenshot(s)
https://github.com/CenterForOpenScience/ember-osf-web/assets/6776190/0bddf490-488e-4d1f-bae0-19e88b2fd264

user with affiliation:
Screenshot 2023-10-19 at 13 19 03

z-index bug (before fix):

https://github.com/CenterForOpenScience/ember-osf-web/assets/6776190/1588b69e-991d-4907-9321-735800cc6854

## QA Notes

[ENG-4814]: https://openscience.atlassian.net/browse/ENG-4814?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/search-page/filter-facet/component.ts [hunk 1]
```diff
@@ -48,12 +48,20 @@ export default class FilterFacet extends Component<FilterFacetArgs> {
     @tracked hasMoreValueOptions = false;
     @tracked nextPageCursor = '';
 
+    get shouldShowTopValues() {
+        return !this.args.property.propertyPathKey.includes('creator');
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/search-page/filter-facet/component.ts:52]: Would this break for preprint's institution affiliation filter?

Comment (aaxelb) [reply lib/osf-components/addon/components/search-page/filter-facet/component.ts:52]: oh, i forgot about that one... yeah i'll make it check `!== 'creator'`, is fine

File: lib/osf-components/addon/components/search-page/component.ts [hunk 2]
```diff
@@ -240,7 +240,7 @@ export default class SearchPage extends Component<SearchArgs> {
             }
             this.filterQueryObject = filterQueryObject;
             const searchResult = await this.store.queryRecord('index-card-search', {
-                cardSearchText,
+                'cardSearchText[*,creator.name]': cardSearchText,
```
[the remark this answers is not in the document]
Comment (aaxelb) [reply lib/osf-components/addon/components/search-page/component.ts:243]: can; api supports it -- i'm already a bit skeptical of the value of `creator.name` (what with lots of false-positive fuzzy matches) and it seems like `isContainedBy.creator.name` might be even worse... but sure, why not

**Review by futa-ikeda — APPROVED** [review]
🔍

File: lib/osf-components/addon/components/search-page/filter-facet/value-option/template.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div local-class={{if @isSelected 'SelectedValueOption'}}>
+    <div>{{@option.indexCard.label}} ({{@option.cardSearchResultCount}})</div>
+    {{#if this.additionalDetail}}
+        <div local-class='AdditionalDetail'>{{this.additionalDetail}}</div>
+    {{/if}}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/search-page/filter-facet/value-option/template.hbs:5]: This is a good addition! 🥞

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**